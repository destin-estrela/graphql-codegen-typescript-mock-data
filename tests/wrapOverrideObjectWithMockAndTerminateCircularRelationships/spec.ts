import { plugin } from '../../src';
import testSchema from './schema';

// TODO: add actual looping types to improve test

it('should support wrapOverrideObjectWithMock and terminateCircularRelationships at the same time', async () => {
    const result = await plugin(testSchema, [], {
        wrapOverrideObjectWithMock: true,
        terminateCircularRelationships: true,
    });

    expect(result).toBeDefined();

    expect(result).toContain(
        "camelCaseThing: overrides && overrides.hasOwnProperty('camelCaseThing') ? (relationshipsToOmit.has('CamelCaseThing') ? overrides.camelCaseThing! : aCamelCaseThing(overrides.camelCaseThing!, relationshipsToOmit)) : relationshipsToOmit.has('CamelCaseThing') ? {} as CamelCaseThing : aCamelCaseThing({}, relationshipsToOmit),",
    );

    expect(result).toContain(
        "listTest: overrides && overrides.hasOwnProperty('listTest') ? overrides.listTest!.map(item => (relationshipsToOmit.has('CamelCaseThing') ? item : aCamelCaseThing(item, relationshipsToOmit))) : [relationshipsToOmit.has('CamelCaseThing') ? {} as CamelCaseThing : aCamelCaseThing({}, relationshipsToOmit)],",
    );

    expect(result).toMatchSnapshot();
});
