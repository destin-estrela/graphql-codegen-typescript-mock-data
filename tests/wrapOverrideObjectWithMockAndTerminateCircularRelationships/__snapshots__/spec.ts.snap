// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should support wrapOverrideObjectWithMock and terminateCircularRelationships at the same time 1`] = `
"
export const aCamelCaseThing = (overrides?: Partial<CamelCaseThing>, _relationshipsToOmit: Set<string> = new Set()): CamelCaseThing => {
    const relationshipsToOmit: Set<string> = new Set(_relationshipsToOmit);
    relationshipsToOmit.add('CamelCaseThing');
    return {
        id: overrides && overrides.hasOwnProperty('id') ? overrides.id! : 'ratione',
    };
};

export const aMain = (overrides?: Partial<Main>, _relationshipsToOmit: Set<string> = new Set()): Main => {
    const relationshipsToOmit: Set<string> = new Set(_relationshipsToOmit);
    relationshipsToOmit.add('Main');
    return {
        camelCaseThing: overrides && overrides.hasOwnProperty('camelCaseThing') ? (relationshipsToOmit.has('CamelCaseThing') ? overrides.camelCaseThing! : aCamelCaseThing(overrides.camelCaseThing!, relationshipsToOmit)) : relationshipsToOmit.has('CamelCaseThing') ? {} as CamelCaseThing : aCamelCaseThing({}, relationshipsToOmit),
        listTest: overrides && overrides.hasOwnProperty('listTest') ? overrides.listTest!.map(item => (relationshipsToOmit.has('CamelCaseThing') ? item : aCamelCaseThing(item, relationshipsToOmit))) : [relationshipsToOmit.has('CamelCaseThing') ? {} as CamelCaseThing : aCamelCaseThing({}, relationshipsToOmit)],
    };
};
"
`;
